lib/resty/http.lua

```diff
@@ -429,7 +435,18 @@ end
 _M.transfer_encoding_is_chunked = transfer_encoding_is_chunked
 
 
-local function _chunked_body_reader(sock, default_chunk_size)
+local function _reader_keepalive_ready_mark(http_client)
+    return co_wrap(function()
+        http_client.keepalive_ready = true
+    end)
+end
+
+local function _reader_keepalive_ready_no_op()
+    return co_wrap(function() end)
+end
+
+
+local function _chunked_body_reader(keepalive_ready_callback, sock, default_chunk_size)
```

Comment:
I'm wondering if we can make this simpler, instead of having to pass in a callback. I guess we could have the readers capture `self` instead, so that they have access to `sock` and the keepalive flags directly?

Or, perhaps the keepalive flags could live in a `reader_state` table, so the readers get `sock`, `reader_state` and `default_chunk_size` to capture?

I appreciate you were probably trying to just slot this in around the current implementation, but I feel like having closures (or worse, more coroutines) just to update a flag is a little overkill.

For context, I regret the liberal use of coroutines in the library in the first place. It would be harder to write, but likely quite a bit more efficient to reimplement without them. A Lua coroutine in OpenResty is essentially a whole new `light thread` - basically an Nginx request context.